library/ssl_client.c

```diff
@@ -598,7 +606,7 @@ static int ssl_write_client_hello_body(mbedtls_ssl_context *ssl,
 #endif
 
 #if defined(MBEDTLS_ECDH_C) || defined(MBEDTLS_ECDSA_C) || \
-    defined(MBEDTLS_KEY_EXCHANGE_ECJPAKE_ENABLED)
+    defined(MBEDTLS_KEY_EXCHANGE_ECJPAKE_ENABLED) || defined(PSA_WANT_ALG_FFDH)
```

Comment:
Looking at this again, I'd say, this should rather be:
```suggestion
    defined(MBEDTLS_KEY_EXCHANGE_ECJPAKE_ENABLED) ||
    (defined(MBEDTLS_SSL_PROTO_TLS1_3) && defined(PSA_WANT_ALG_FFDH))
```
ditto line 188.